### PR TITLE
Add basic Dockerfile

### PR DESCRIPTION
**Pull request by adamdecaf**
I didn't bother setting up a non-default config.json or provide valid credentials, but this image gathers everything up together.

```
$ docker build .
Sending build context to Docker daemon  818.7kB
Step 1/3 : FROM python:2.7-onbuild
...
Successfully built 62b2b218ce77

$ docker run -it 62b2b218ce77 schedule_repo foo
Loaded config from /usr/src/app/config.json
INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 169.254.169.254
Traceback (most recent call last):
  File "/usr/local/bin/repokid", line 11, in <module>
    load_entry_point('repokid', 'console_scripts', 'repokid')()
  File "/usr/local/lib/python2.7/site-packages/repokid/cli/repokid_cli.py", line 950, in main
    dynamo_table = dynamo_get_or_create_table(**config['dynamo_db'])
  File "/usr/local/lib/python2.7/site-packages/repokid/utils/dynamo.py", line 64, in dynamo_get_or_create_table
    region=dynamo_config['region'])
  File "/usr/local/lib/python2.7/site-packages/cloudaux/aws/decorators.py", line 35, in decorated_function
    retval = f(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/cloudaux/aws/sts.py", line 96, in boto3_cached_conn
    role = sts.assume_role(RoleArn=arn, RoleSessionName=session_name)
  File "/usr/local/lib/python2.7/site-packages/botocore/client.py", line 317, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python2.7/site-packages/botocore/client.py", line 602, in _make_api_call
    operation_model, request_dict)
  File "/usr/local/lib/python2.7/site-packages/botocore/endpoint.py", line 143, in make_request
    return self._send_request(request_dict, operation_model)
  File "/usr/local/lib/python2.7/site-packages/botocore/endpoint.py", line 168, in _send_request
    request = self.create_request(request_dict, operation_model)
  File "/usr/local/lib/python2.7/site-packages/botocore/endpoint.py", line 152, in create_request
    operation_name=operation_model.name)
  File "/usr/local/lib/python2.7/site-packages/botocore/hooks.py", line 227, in emit
    return self._emit(event_name, kwargs)
  File "/usr/local/lib/python2.7/site-packages/botocore/hooks.py", line 210, in _emit
    response = handler(**kwargs)
  File "/usr/local/lib/python2.7/site-packages/botocore/signers.py", line 90, in handler
    return self.sign(operation_name, request)
  File "/usr/local/lib/python2.7/site-packages/botocore/signers.py", line 156, in sign
    auth.add_auth(request)
  File "/usr/local/lib/python2.7/site-packages/botocore/auth.py", line 352, in add_auth
    raise NoCredentialsError
botocore.exceptions.NoCredentialsError: Unable to locate credentials
```

Issue: https://github.com/Netflix/repokid/issues/42

### CODE REVIEW

**Comment by adamdecaf** [comment]
Do I need to change something after moving the requirements out of `setup.py`? I'm not very familiar with python.

File: setup.py [hunk 1]
```diff
@@ -25,18 +25,6 @@
...
-        'cloudaux==1.2.0',
-        'docopt==0.6.2',
-        'import_string==0.1.0',
-        'marshmallow==2.13.5',
-        'policyuniverse',
-        'requests==2.13.0',
-        'tabulate==0.7.7',
-        'tabview==1.4.2',
-        'tqdm==4.11.2'
-    ],
```
Comment (castrapel) [inline setup.py]: Would you mind changing this to something like the following?

`with open('requirements.txt') as f:`
 `   REQUIRED = f.read().splitlines()`
...
`setup(`
...
`
    install_requires=REQUIRED,`
...`

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+boto3
```
Comment (castrapel) [inline requirements.txt]: Let's take this opportunity to fix these dependencies and make them easier to update. Would you mind making a requirements.in file with he high level (unpinned) dependencies, and running `pip-compile --output-file requirements.txt requirements.in` to create the requirements.txt file with the pinned dependencies?

More context: https://github.com/jazzband/pip-tools

Comment (adamdecaf) [reply requirements.txt]: I'm having trouble running this command. Do I need to setup a python virutalenv outside of the project and use that?

Comment (castrapel) [reply requirements.txt]: Yes, you can do that. `pip install pip-tools` should install `pip-compile` for you. 

$ virtualenv env2
New python executable in /Users/ccastrapel/localrepos/consoleme/env2/bin/python
Installing setuptools, pip, wheel...done.
$ . env2/bin/activate
$ pip install pip-tools
Collecting pip-tools
  Using cached pip_tools-1.11.0-py2.py3-none-any.whl
Collecting first (from pip-tools)
  Using cached first-2.0.1-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./env2/lib/python2.7/site-packages (from pip-tools)
Collecting six (from pip-tools)
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting click>=6 (from pip-tools)
  Using cached click-6.7-py2.py3-none-any.whl
Installing collected packages: first, six, click, pip-tools
Successfully installed click-6.7 first-2.0.1 pip-tools-1.11.0 six-1.11.0
$ pip-compile --output-file requirements.txt requirements.in

**Comment by adamdecaf** [comment]
@castrapel Should be g2g now. Thanks!

**Comment by mcpeak** [comment]
@adamdecaf if you hop over to Gitter and PM me I'd like to send you something small as a thank you.

**Pull request merged**